### PR TITLE
[BugFix] when struct in array append_default real struct row for noexist subfiled

### PR DESCRIPTION
**Pull request by zombee0**
Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 11 / 31 (35.48%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/formats/parquet/group_reader.cpp|0|4|00.00%|[541, 551, 552, 553]|
|:large_blue_circle:|src/formats/parquet/column_reader.cpp|11|27|40.74%|[740, 741, 746, 750, 751, 752, 753, 754, 763, 765, 768, 769, 770, 771, 772, 827]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 11 / 31 (35.48%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/formats/parquet/group_reader.cpp|0|4|00.00%|[541, 551, 552, 553]|
|:large_blue_circle:|src/formats/parquet/column_reader.cpp|11|27|40.74%|[740, 741, 746, 750, 751, 752, 753, 754, 763, 765, 768, 769, 770, 771, 772, 827]|

File: be/src/formats/parquet/column_reader.cpp [hunk 1]
```diff
@@ -786,21 +805,36 @@ class StructColumnReader : public ColumnReader {
         DCHECK_EQ(field_names.size(), _child_readers.size());
 
         // Fill data for subfield column reader
+        size_t real_read = 0;
```
[the remark this answers is not in the document]
Comment (zombee0) [reply be/src/formats/parquet/column_reader.cpp:808]: it does not means row num...

File: be/src/formats/parquet/column_reader.cpp [hunk 2]
```diff
@@ -737,23 +737,42 @@ class StructColumnReader : public ColumnReader {
...
-            Column* child_column = struct_column->field_column(field_name).get();
             if (LIKELY(_child_readers.find(field_name) != _child_readers.end())) {
                 if (_child_readers[field_name] != nullptr) {
+                    Column* child_column = struct_column->field_column(field_name).get();
                     _opts.context->next_row = origin_next_row;
                     _opts.context->rows_to_skip = origin_rows_to_skip;
                     RETURN_IF_ERROR(_child_readers[field_name]->prepare_batch(num_records, child_column));
-                } else {
-                    child_column->append_default(*num_records);
+                    size_t current_real_read = child_column->size();
```
[the remark this answers is not in the document]
Comment (zombee0) [reply be/src/formats/parquet/column_reader.cpp:750]: it does not mean row num...

**Review by Smith-Cruise — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**